jaxrs-tck/src/main/java/jakarta/ws/rs/tck/jaxrs31/spec/extensions/JAXRSClientIT.java

```diff
@@ -0,0 +1,120 @@
+ */
+
+package jakarta.ws.rs.tck.jaxrs31.spec.extensions;
+
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.container.DynamicFeature;
+import jakarta.ws.rs.core.Feature;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.tck.common.client.JaxrsCommonClient;
+import jakarta.ws.rs.tck.common.client.JdkLoggingFilter;
+import jakarta.ws.rs.tck.jaxrs21.spec.completionstage.CompletionStageResource;
+import jakarta.ws.rs.tck.lib.util.TestUtil;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit5.ArquillianExtension;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.io.IOException;
+import java.util.concurrent.Future;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+/*
+ * @class.setup_props: webServerHost;
+ *                     webServerPort;
+ */
+@ExtendWith(ArquillianExtension.class)
+public class JAXRSClientIT extends JaxrsCommonClient {
+
+  private static final long serialVersionUID = 31L;
+
+  public JAXRSClientIT() {
+    setup();
+    setContextRoot("/jaxrs_jaxrs31_spec_jdkservices_web");
+  }
+
+  @BeforeEach
+  void logStartTest(TestInfo testInfo) {
+    TestUtil.logMsg("STARTING TEST : "+testInfo.getDisplayName());
+  }
+
+  @AfterEach
+  void logFinishTest(TestInfo testInfo) {
+    TestUtil.logMsg("FINISHED TEST : "+testInfo.getDisplayName());
+  }
+
+
+  @Deployment(testable = false)
+  public static WebArchive createDeployment() throws IOException{
+    StringAsset dynamicFeatureServiceFile = new StringAsset(TckDynamicFeature.class.getName());
+    StringAsset featureServiceFile = new StringAsset(TckFeature.class.getName());
+    String prefix = "classes/META-INF/services/";
+
+    WebArchive archive = ShrinkWrap.create(WebArchive.class, "jaxrs_jaxrs31_spec_jdkservices_web.war");
+    archive.addClasses(TSAppConfig.class, DynamicFeatureResource.class,
+            TckFeature.class, TckFeature.TckFeatureFilter.class,
+            TckDynamicFeature.class, TckDynamicFeature.TckDynamicFeatureFilter.class);
+    archive.addAsWebInfResource(dynamicFeatureServiceFile, prefix + DynamicFeature.class.getName());
+    archive.addAsWebInfResource(featureServiceFile, prefix + Feature.class.getName());
+    return archive;
+
+  }
+
+  /* Run test */
+
+  /*
+   * @testName: featureIsRegisteredTest
+   * 
+   * @assertion_ids: JAXRS:SPEC:137;
+   * 
+   * @test_Strategy:
+   */
+  @Test
+  public void featureIsRegisteredTest() throws Fault {
+    try (Response response = ClientBuilder.newClient().target(getAbsoluteUrl("staticFeature")).request().get()) {
```

Comment:
I think it would be possible to do something like the following here:
```
setProperty(REQUEST, buildRequest(GET, "staticFeature"));
setProperty(SEARCH_STRING, TckFeature.class.getName());
invoke();
```
It would be shorter, yes, but I don't find that particularly beautiful so I would be fine with leaving it like it is.